src/opm/output/eclipse/RestartIO.cpp

```diff
@@ -537,43 +536,38 @@ void writeWell(ecl_rst_file_type* rst_file, int sim_step, const EclipseState& es
     write_kw( rst_file, ERT::EclKW< int >( ICON_KW, icon_data ) );
 }
 
-void checkSaveArguments(const data::Solution& cells,
-                        const EclipseGrid& grid,
-                        const std::map<std::string, std::vector<double>>& extra_data) {
-
-    const std::set<std::string> reserved_keys = {"LOGIHEAD", "INTEHEAD" ,"DOUBHEAD", "IWEL", "XWEL","ICON", "XCON" , "OPM_IWEL" , "OPM_XWEL", "ZWEL"};
+void checkSaveArguments(const EclipseState& es,
+                        const RestartValue& restart_value,
+                        const EclipseGrid& grid) {
 
-    for (const auto& pair : extra_data) {
-        const std::string& key = pair.first;
-        if (key.size() > 8)
-            throw std::runtime_error("The keys in extra data must have maximum eight characaters");
+  for (const auto& elm: restart_value.solution)
+    if (elm.second.data.size() != grid.getNumActive())
+      throw std::runtime_error("Wrong size on solution vector: " + elm.first);
 
-        if (cells.has( key ))
-            throw std::runtime_error("The keys used must unique across Solution and extra_data");
 
-        if (reserved_keys.find( key ) != reserved_keys.end())
-            throw std::runtime_error("The extra_data uses a reserved key");
-    }
+  if (es.getSimulationConfig().getThresholdPressure().size() > 0) {
+      if (!restart_value.has_extra("THPRES"))
+          throw std::runtime_error("This model has THPRES active - must have THPRES as part of restart data.");
```

Comment:
This breaks flow_legacy (family) on Norne (for example), since only flow(ebos) has been updated to provide these values. We could reduce this to an OpmLog::warning() to get around that. I acknowledge that its days are counted, but as of now we still have legacy-family codes that are active.

Reply:
> This breaks flow_legacy (family) on Norne

Yes; and I think it is not worth it to fix in opm-simulators.

Reply:
I agree, but I thought that we therefore also agreed on the workaround here (to reduce to warning to allow legacy to run with minimal effort), but when I look at the diff now it seems to me this was not changed?

Reply:
Ehhh no; we did not agree on that. I want the code to remain exactly like it is - that enforces correctness. If we avoid enforcing this correctness, we actually let the not-maintaned-state of flow legacy leak, I think that is a wrong decision.

Reply:
But then we must either remove the legacy simulators *now* (which I don't think is feasible in particular due to the sequential ones having no replacements) or do the work required to have THPRES info from the legacy sims, which is what you wanted to avoid. I am fine with that, but it will of course be a little more work, delaying this PR chain.

Reply:
> But then we must either remove the legacy simulators now (which I don't think is feasible in particular due to the sequential ones having no replacements) or do the work required to have THPRES info from the legacy sims

No - we can just let it slide; the fact that the legacy simulators can not restart a simulation with `THPRES`enabled is not a significant regression - but I will fix it.

Reply:
The issue is not that it cannot restart with THPRES (that I can live with), but that it will not run at all. The first output call throws:
```
...
Keyword WTPTSEA is unhandled
Keyword WTPTTFB is unhandled


================ Starting main simulation loop ===============


Error: Program threw an exception: This model has THPRES active - must have THPRES as part of restart data.

```